File: src/main/java/net/explorviz/adapter/service/converter/SpanConverter.java
package net.explorviz.adapter.service.converter;

import io.opencensus.proto.trace.v1.Span;

/**
 * Converts OpenCensus spans into another type.
 * @param <T> the type spans are converted to.
 */
public interface SpanConverter<T> {

  /**
   * Converts an OpenCensus {@link Span} into {@link T}.
   */
  T fromOpenCensusSpan(Span ocSpan);

}

File: src/main/java/net/explorviz/adapter/service/converter/SpanConverter.kt
package net.explorviz.adapter.service.converter;

import io.opentelemetry.proto.trace.v1.Span;

/**
 * Converts OpenCensus spans into another type.
 *
 * @param <T> the type spans are converted to.
 */
interface SpanConverter<T> {

  /**
   * Converts an OpenCensus {@link Span} into {@link T}.
   */
  fun fromOpenTelemetrySpan(ocSpan: Span): T;

}
